src/components/forms/JurisdictionMetadata/index.tsx

```diff
@@ -0,0 +1,230 @@
+import * as Yup from 'yup';
+import {
+  CSV_FILE,
+  FILE,
+  FILE_UPLOADED_SUCCESSFULLY,
+  JURISDICTION_ID,
+  JURISDICTION_METADATA,
+  JURISDICTION_NAME,
+  REQUIRED,
+  UPLOAD,
+  UPLOAD_FILE,
+  UPLOADING,
+} from '../../../configs/lang';
+import { HOME_URL, OPENSRP_V1_SETTINGS_ENDPOINT } from '../../../constants';
+import { growl } from '../../../helpers/utils';
+import { OpenSRPService } from '../../../services/opensrp';
+
+const SUPPORTED_FORMATS = ['text/csv'];
+
+/** yup validation schema for Jurisdiction Metadata Form input */
+export const JurisdictionSchema = Yup.object().shape({
+  file: Yup.mixed()
+    .required(REQUIRED)
+    .test('fileFormat', 'CSV Files only', value => value && SUPPORTED_FORMATS.includes(value.type)),
+});
+
+export interface JurisdictionMetadataFormFields {
+  file: File;
+}
+
+export interface Setting {
+  description: string;
+  label: string;
+  value: string | unknown;
+  key: string;
+}
+
+export interface SettingConfiguration {
+  type: string;
+  identifier: string;
+  providerId: string;
+  locationId: string;
+  settings: Setting[];
+  teamId: string;
+}
+
+export interface JurisdictionMetadataFormProps {
+  disabledFields: string[];
+  OpenSRPService: new (...args: any[]) => any;
+  submitForm: (
+    setSubmitting: (isSubmitting: boolean) => void,
+    setGlobalError: (errorMessage: string) => void,
+    setIfDoneHere: (closeSubmissionCycle: boolean) => void,
+    props: JurisdictionMetadataFormProps,
+    values?: JurisdictionMetadataFormFields
+  ) => void;
+  initialValues: JurisdictionMetadataFormFields;
+  redirectAfterAction: string;
+}
+
+export const defaultInitialValues: JurisdictionMetadataFormFields = {
+  file: new File([], ''),
+};
+
+// create payloads for sending
+const createPayloads = (result: any): SettingConfiguration[] => {
+  const payloads: SettingConfiguration[] = [];
+  const { data } = result;
+  // check if jurisdiction_id exists
+  if (data.length > 0 && data[0].jurisdiction_id) {
+    // get the metadata items
+    const headers = Object.keys(data[0]);
+    for (const header of headers) {
+      const settings: Setting[] = [];
+      if (header !== JURISDICTION_ID && header !== JURISDICTION_NAME) {
+        // add the metadata values with jurisdiction as the key
+        for (const item of data) {
+          const entries = Object.entries(item);
+          for (const [key, value] of entries) {
+            if (key === header) {
+              const setting: Setting = {
+                description: `${JURISDICTION_METADATA} for ${item.jurisdiction_name} id ${item.jurisdiction_id}`,
+                key: item.jurisdiction_id,
+                label: `${item.jurisdiction_name} metadata`,
+                value,
+              };
+              settings.push(setting);
+            }
+          }
+        }
+        const payload: SettingConfiguration = {
+          identifier: `jurisdiction_metadata-${header}`,
+          locationId: '',
+          providerId: 'demo',
```

Comment:
Let us use the currently logged in user for this.  There is a `getUser` selector from '@onaio/session-reducer';